### PR TITLE
Use IClientAssertionCertificate instead

### PR DESCRIPTION
**Pull request by ChenTanyi**
After my testing, there is issue in `ClientAssertionCertificate` to sign certificate. https://github.com/Azure/azure-libraries-for-net/pull/1041
I want to use the `IdentityModel.Clients.ActiveDirectory.ClientAssertionCertificate`, so I change the parameter to `IClientAssertionCertificate` instead

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
@ChenTanyi Any detail of the problem on ClientAssertionCertificate? It might help auth sdk fix that class.

I didn't find source code from Microsoft.IdentityModel.Clients.ActiveDirectory.ClientAssertionCertificate

**Comment by ChenTanyi** [comment]
https://github.com/Azure/azure-sdk-for-net/blob/45cdad6bc3f20b36220d303a44f99895bdba28b7/sdk/mgmtcommon/Auth/Az.Auth/Az.Authentication/ClientAssertionCertificate.cs#L44
https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/68d7dea3643e075be85abbca1ab88ba614465541/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Features/CertificateAssertion/ClientAssertionCertificate.cs#L101

This two signature algorithm is different. One is successful for the `sendx5c` certificate and the other is failed after my testing. I don't know the reason.

The error in oauth is
```
AADSTS700027: Client assertion contains an invalid signature. [Reason - The key was not found., Thumbprint of key used by client: '***', Please visit 'https://developer.microsoft.com/en-us/graph/graph-explorer' and query for 'https://graph.microsoft.com/beta/applications/***' to see configured keys]
```

File: sdk/mgmtcommon/Auth/Az.Auth/Az.Authentication/CertificateAuthenticationProvider.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ public CertificateAuthenticationProvider(string certificateFilePath, bool IsCert
             _assertionProvider = (clientId) => Task.FromResult<ClientAssertionCertificate>(new ClientAssertionCertificate(clientId, certificateFilePath));
         }
 
-        public CertificateAuthenticationProvider(ClientAssertionCertificate certAssertion, bool IsCertRollOverEnabled) : this()
+        public CertificateAuthenticationProvider(IClientAssertionCertificate certAssertion, bool IsCertRollOverEnabled) : this()
```
Comment (yungezz) [inline sdk/mgmtcommon/Auth/Az.Auth/Az.Authentication/CertificateAuthenticationProvider.cs:68]: how the change fixing the problem? besides `ClientAssertionCertificate` any other impl of the interface `IClientAssertionCertificate`?

Comment (ChenTanyi) [reply sdk/mgmtcommon/Auth/Az.Auth/Az.Authentication/CertificateAuthenticationProvider.cs:68]: For how, you can see the PR in `libraries-for-net`. We could use other implementation as parameter. The implementation is in the original repo in aad. I have commented above.

**Pull request closed without merging**